events-generator/src/main/java/org/moditect/jfrunit/generator/JfrUnitConstantsGenerator.java

```diff
@@ -0,0 +1,130 @@
+package org.moditect.jfrunit.generator;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.Writer;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.tools.JavaFileObject;
+
+import org.moditect.jfrunit.generator.events.model.Event;
+import org.moditect.jfrunit.generator.events.model.JfrDoc;
+import org.moditect.jfrunit.generator.events.model.Type;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import freemarker.template.TemplateExceptionHandler;
+
+public class JfrUnitConstantsGenerator {
+
+    static List<String> baseTypes = new ArrayList<>();
+    static ObjectMapper MAPPER = new ObjectMapper();
+    static Logger LOGGER = LoggerFactory.getLogger(JfrUnitConstantsGenerator.class);
+    static String PACKAGE = "org.moditect.jfrunit.events";
+    static String BASE_FOLDER_GEN = "src/main/java/";
+    private static Configuration cfg = new Configuration(Configuration.VERSION_2_3_31);
+
+    public static void generate(String jfrDocUrl, ProcessingEnvironment processingEnvironment) throws IOException, TemplateException {
+        if (jfrDocUrl == null) {
+            String error = "missing jfr-doc json argument (ex https://bestsolution-at.github.io/jfr-doc/openjdk-17.json)";
+            LOGGER.error(error);
+            throw new IllegalArgumentException(error);
+        }
+
+        JfrDoc jrfDoc = MAPPER.readValue(new BufferedInputStream(new URL(jfrDocUrl).openStream()), JfrDoc.class);
+
+        LOGGER.info("generating sources for version {} and distribution {}", jrfDoc.getVersion(), jrfDoc.getDistribution());
+
+        configureFremarker();
+
+        generateEventClasses(processingEnvironment, jrfDoc);
+
+        generateJfrEventTypes(processingEnvironment, jrfDoc);
+
+        generateTypeClasses(processingEnvironment, jrfDoc);
+
+        LOGGER.info("sources generated in {} folder", BASE_FOLDER_GEN);
+    }
+
+    private static void generateTypeClasses(ProcessingEnvironment processingEnvironment, JfrDoc jrfDoc) throws IOException, TemplateException {
```

Comment:
A comment on each of those methods would be nice, so to clarify the difference between event classes, event types, and type classes.